src/database/indexer.rs

```diff
@@ -97,7 +97,17 @@ fn update_repository_metadata(scan_path: &Path, repository_list: Option<&Path>,
 }
 
 fn find_default_branch(repo: &gix::Repository) -> Result<Option<String>, anyhow::Error> {
-    Ok(Some(repo.head()?.name().as_bstr().to_string()))
+    if repo.head()?.is_detached() {
+        Ok(None)
```

Comment:
I think this is redundant because `referent_name` already checks if detached and returns None so could've been written:

```rust
let head = repo.head()?;

let Some(referent) = head.referent_name() else {
    return Ok(None);
};

Ok(Some(referent.as_bstr().to_string()))
```

Not a huge issue though. Thanks for the contribution